### DISCUSSION TITLE
Setting headers for presign_url not working

### DISCUSSION THREAD

**Discussion by bigbean17** [aws/aws-sdk-ruby#2916] [Q&A]
Hi
I'm trying to create the presigned url for kinesis signaling channel using "presign_url". I need to set the header of X-Amz-ChannelARN for the presigned url. 

I modified the code mentioned [here](https://github.com/aws/aws-sdk-ruby/blob/204a3b7b43f1619fcd40b395751cca987ddebe40/gems/aws-sigv4/lib/aws-sigv4/signer.rb#L568)

```
 url = signer.presign_url(
  http_method: 'GET',
  url: channel_url,
  headers: {
     'X-Amz-channelARN' => arn
  }
```
But it is not working, the generated url contains something like this:X-Amz-SignedHeaders=host%3Bx-amz-channelarn
Has anyone met the similar issue before?

Thanks~

**Comment by mullermp** [comment]
If the header becomes signed, you will need to include it in your request. It will not be hoisted onto the query string.

**Reply by bigbean17** [reply]
Hi mullermp,

Thanks for the quick reply, but I'm still bit confused of how to add this header to the request. Now i've removed the headers option, so it looks like this 
```
 presigned_url = signer.presign_url(
  http_method: 'GET',
  url: channel_url,
 ```
And the url will be used by the JS viewer to generate the client
```
        client = new KVSWebRTC.SignalingClient({
            channelARN: 'default',
            channelEndpoint:'default',
            clientId: cid,
            role: viewer
            region: region,
            requestSigner: new Signer(presigned_url)
        });
```
This is my Signer
```
class Signer {
    constructor(url){
        this.url = url
    }
    getSignedURL(){
        return this.url
    }
}
```
I'm not really sure where to add that header "X-AMZ-channelARN" to the request. 

Thanks~

**Reply by mullermp** [reply]
You need to have the header signed like you were before, and when you make your http request using the presigned url, you need to include the same header and value that you signed with.
